Demo/Demo/Features/FeatureBaseViewController.swift

```diff
@@ -0,0 +1,103 @@
+        let label = UILabel()
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureBottomView()
+        processOrderButton.isHidden = true
+    }
+
+    func configureBottomView() {
+        view.addSubview(processOrderButton)
+        view.addSubview(titleLabel)
+
+        NSLayoutConstraint.activate([
+            processOrderButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            processOrderButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            processOrderButton.heightAnchor.constraint(equalToConstant: 40),
+
+            titleLabel.topAnchor.constraint(equalTo: processOrderButton.bottomAnchor, constant: 16),
+            titleLabel.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            titleLabel.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            titleLabel.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -16)
+        ])
+    }
+
+    func fetchOrderID(paymentMethod: PaymentMethod) {
+        // TODO: The prod server is broken right now, so we will expect those requests to fail.
+        // TODO: Pass order details into this function
+
+        let amount = Amount(currencyCode: "USD", value: "10.00")
+        let orderRequestParams = CreateOrderParams(
+            intent: DemoSettings.intent.rawValue.uppercased(),
+            purchaseUnits: [PurchaseUnit(amount: amount)]
+        )
+
+        DemoMerchantAPI.sharedService.createOrder(orderParams: orderRequestParams) { result in
+            switch result {
+            case .success(let order):
+                self.processOrder(paymentMethod: paymentMethod, orderID: order.id, state: .orderCreated)
+                DispatchQueue.main.async {
+                    self.processOrderButton.isHidden = false
+                }
+                print("✅ fetched orderID: \(order.id) with status: \(order.status)")
+            case .failure(let error):
+                self.processOrder(paymentMethod: paymentMethod, state: .orderFailed)
+                print("❌ failed to fetch orderID: \(error)")
+            }
+        }
+    }
+
+    func processOrder(paymentMethod: PaymentMethod, orderID: String? = nil, state: OrderState) {
+        let message: String
+
+        switch state {
+        case .notStarted:
+            message = "Enter your \(paymentMethod) details to begin"
+        case .orderProcessing:
+            message = "Processing your \(paymentMethod) order"
+        case .orderCreated:
+            message = "Order ID: \(orderID ?? "")"
+        case .orderFailed:
+            message = "Your order has failed, please try again"
+        }
```

Comment:
Not sure if all these enum cases apply - since the first method above `fetchOrder` is what actually creates this order. This method should submit that order for authorizing or capturing, only needing an orderID string and intent.

Reply:
The state when we call this method could be like `Submitted order for authorize/capture ... `. This method should call `func processOrder(processOrderParams: ProcessOrderParams, completion: @escaping ((Result<Order, URLResponseError>) -> Void))` in our Demo API Client.

Let me know if that makes sense!

Reply:
Updated - removed the enum and switch altogether for now since the labels may be in flux for a bit. We can always add it back in and pass the state into the func in the future if we want.